File: read_pdfs.py
import subprocess
import sys

def install(package):
    subprocess.check_call([sys.executable, "-m", "pip", "install", package])

try:
    import pypdf
except ImportError:
    install('pypdf')
    import pypdf

def read_pdf(file_path):
    try:
        with open(file_path, "rb") as f:
            reader = pypdf.PdfReader(f)
            text = ""
            for page in reader.pages:
                text += page.extract_text() + "\n"
            print(f"=== {file_path} ===")
            print(text)
    except Exception as e:
        print(f"Error reading {file_path}: {e}")

read_pdf(r"C:\Proyecto MiniMercado\Proyecto\arquitectura_pro.pdf")
read_pdf(r"C:\Proyecto MiniMercado\Proyecto\enfoque_pro.pdf")

File: read_pdfs.js
const fs = require('fs');
const pdf = require('pdf-parse');

async function readPdf(filePath) {
    try {
        const dataBuffer = fs.readFileSync(filePath);
        const parseFunc = typeof pdf === 'function' ? pdf : pdf.default;
        const data = await parseFunc(dataBuffer);
        console.log(`\n\n=== CONTENT OF ${filePath} ===\n`);
        console.log(data.text);
    } catch (error) {
        console.error(`Error reading ${filePath}:`, error.message, error.stack);
    }
}

async function main() {
    await readPdf('arquitectura_pro.pdf');
    await readPdf('enfoque_pro.pdf');
}

main();
